src/main/java/apoc/load/LoadXls.java

```diff
@@ -172,17 +180,60 @@ private Object convertArray(Object value) {
 
         private Object convertType(Object value) {
             if (nullValues.contains(value) || value == null) return null;
-//            if (!(value instanceof String)) return value;
             switch (type) {
-                case STRING: return value.toString();
-                case INTEGER: return Util.toLong(value);
-                case FLOAT: return Util.toDouble(value);
-                case BOOLEAN: return Util.toBoolean(value);
-                case NULL: return null;
-                case LIST: return Arrays.stream(arrayPattern.split(value.toString())).map(this::convertType).collect(Collectors.toList());
-                default: return value;
+                case STRING:
+                    if (value instanceof Temporal && dateFormat instanceof String && dateFormat != null) {
```

Comment:
- togliamo `dateFormat instanceof String`
- da una parte fai `value instanceof Temporal` e poi `(TemporalAccessor) value`, quando fai l'`instanceof`di una classe poi dei castare l'oggetto alla stessa classe dell'`instaceof`
- creiamo un metodo dentro Util `Util.formatDate((TemporalAccessor) value, dateFormat.toString())`
dovrebbe diventare:
```suggestion
                    if (value instanceof Temporal && && dateFormat != null) {
                              return Util.formatDate(value,
                    }
```